07 - Arrays/Program.cs

```diff
@@ -5,7 +5,179 @@ namespace _07___Arrays
+                }
+                // Diagonal from the right
+                else if (ticTacToe[0, 2] == ticTacToe[1, 1] && ticTacToe[1, 1] == ticTacToe[2, 0])
+                {
+                    Console.WriteLine(winMessage, ticTacToe[0, 2]);
+                }
+                // Last horizontal
+                else if (ticTacToe[2, 0] == ticTacToe[2, 1] && ticTacToe[2, 1] == ticTacToe[2, 2])
+                {
+                    Console.WriteLine(winMessage, ticTacToe[2, 0]);
+                }
+                // middle horizontal
+                else if (ticTacToe[1, 0] == ticTacToe[1, 1] && ticTacToe[1, 1] == ticTacToe[1, 2])
+                {
+                    Console.WriteLine(winMessage, ticTacToe[1, 0]);
+                }
+                // upper horizontal
+                else if (ticTacToe[0, 0] == ticTacToe[0, 1] && ticTacToe[0, 1] == ticTacToe[0, 2])
+                {
+                    Console.WriteLine(winMessage, ticTacToe[0, 0]);
+                }
+                // Down first
+                else if (ticTacToe[0, 0] == ticTacToe[1, 0] && ticTacToe[1, 0] == ticTacToe[2, 0])
+                {
+                    Console.WriteLine(winMessage, ticTacToe[0, 0]);
+                }
+                // Down middle
+                else if (ticTacToe[0, 1] == ticTacToe[1, 1] && ticTacToe[1, 1] == ticTacToe[2, 1])
+                {
+                    Console.WriteLine(winMessage, ticTacToe[0, 1]);
+                }
+                // Down last
+                else if (ticTacToe[0, 2] == ticTacToe[1, 2] && ticTacToe[1, 2] == ticTacToe[2, 2])
+                {
+                    Console.WriteLine(winMessage, ticTacToe[0, 2]);
+                }
+                else
+                {
+                    ChooseSquare();
+                }
+            }
+
+            void SetCrossOrCircle(int value)
+            {
+                string stringValue = value.ToString();
+
+                for (int y = 0; y < 3; y++)
+                {
+                    for (int x = 0; x < 3; x++)
+                    {
+                        if (ticTacToe[x, y] == stringValue)
+                        {
+                            if (turnForX.Contains(turnNumber))
+                            {
+                                ticTacToe[x, y] = "X";
+                                Console.Clear();
+                                ShowDeck();
+                                CheckForWin();
+                            }
+                             else
+                            {
+                                ticTacToe[x, y] = "O";
+                                Console.Clear();
+                                ShowDeck();
+                                CheckForWin();
+                            }
+                           
+                        }
+                    }
+                }
+            }
+
+            bool CheckifSquareIsTaken(int squareNum)
```

Comment:
not square - cell

Reply:
Square is from board games terminology. Not sure what would be better in this context